jenkinsfiles/stable

```diff
@@ -252,14 +253,31 @@ pipeline {
                             }
 
                             dir('scripts/databases') {
-                                env.DATABASE_ID = sh(
-                                        returnStdout: true,
-                                        script: "./list.sh | jq -r '.[] | select(.name == \"test-dbs\") | .databases[] | select(.name == \"sierra-leone/${version}.sql.gz\") | .slug'"
-                                ).trim()
-                            }
+                                DATABASE_GROUP_NAME = 'test-dbs'
+                                env.DATABASE_ID = im.findDatabaseId(DATABASE_GROUP_NAME, version)
+
+                                if (!env.DATABASE_ID) {
+                                    echo "Couldn't find database for $version"
+
+                                    try {
+                                        env.DATABASE_ID = im.uploadNewDatabase(DATABASE_GROUP_NAME, version)
+                                    } catch (err) {
+                                        echo "Couldn't download database for ${version}: ${err}"
+
+                                        shortVersion = version.split('\\.').take(2).join('.')
+                                        env.DATABASE_ID = im.findDatabaseId(DATABASE_GROUP_NAME, shortVersion)
 
-                            sh '[ -n "$DATABASE_ID" ]'
-                            echo "Database: ${env.DATABASE_ID}"
+                                        if (!env.DATABASE_ID) {
+                                            echo "Couldn't find database for $shortVersion"
+
+                                            env.DATABASE_ID = im.uploadNewDatabase(DATABASE_GROUP_NAME, shortVersion)
```

Comment:
should we catch any errors here? - (just to handle them gracefully)

_not sure it's necessary, as it is quite unlikely to reach here_

Reply:
If we can't find a database even by the "short version", like 2.41, then the pipeline should fail I think.

Reply:
Yes, agreed. 

Will that just "happen" with the current implementation, or should we throw an error here and exit?

Reply:
Yeah, the uploadNewDatabase function will fail if there's no database to download for the given version.